edward/inferences/inference.py

```diff
@@ -258,3 +284,48 @@ def finalize(self):
     """
     if self.logging:
       self.train_writer.close()
+
+  def set_log_variables(self, logvars=None, log_max_scalers_per_var=None):
+    """Logs variables to TensorBoard
+
+     For each variable in logvars, creates 'scalar' and / or 'histogram' by
+     calling `tf.summary.scalar` or `tf.summary.histogram`
+
+     if logvars is None, automatically log all latent variables that have been
+     given non-default names.  If logvars is [], no logging will be created.
+
+     Parameters
+     ----------
+     logvars : list, optional
+       A list of variables to be logged
+     log_max_scalers_per_var : int, default None
+       Enables logging of individual values from 1 dimensional variables up to
+       a maximum dimension, if None will log all dimensions.
+
+     Returns
+     -------
+     None
+
+    """
+    if logvars is None:
+      logvars = []
+      for k in self.latent_vars:
+        logvars += get_variables(self.latent_vars[k])
+
+      # Prune variables to only be custom named variables (without 'Variable')
+      # substring
+      logvars = [var for var in logvars if 'Variable' not in var.name]
+
+    for var in logvars:
+      var_name = var.name.replace(':', '/')  # colons are an invalid character
+
+      # If variable is a one dimensional tensor, log each element in the tensor
+      # individually. Only log the first log_max_scalers_per_var variables
+      if len(var.shape) == 1:
+        for i in range(var.shape[0]):
+          if log_max_scalers_per_var is None or i < log_max_scalers_per_var:
+            tf.summary.scalar('{}/{}'.format(var_name, i), var[i])
+
+      # If var is multi-dimensional, log the distribution
+      if len(var.shape) > 0 and np.max(var.shape) > 1:
+        tf.summary.histogram(var_name, var)
```

Comment:
Some of this conflicts with the current tensorboard tracking in `variational_inference.py` (see #654). My guess is you can remove the parameter histograms in `variational_inference.py`. 

I also recommend checking to see how the naming systems match.